androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/SeekClickableArea.kt

```diff
@@ -0,0 +1,72 @@
+package com.profusion.androidenhancedvideoplayer.components.playerOverlay
+
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.indication
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import com.profusion.androidenhancedvideoplayer.R
+import com.profusion.androidenhancedvideoplayer.styling.Dimensions
+
+@Composable
+fun SeekClickableArea(
+    modifier: Modifier = Modifier,
+    scaleAnimation: Float,
+    tapCount: Int,
+    onSeekSingleTap: () -> Unit,
+    onSeekDoubleTap: () -> Unit,
+    checkIfCanToggleIsControlsVisible: () -> Unit,
+    getSeekTime: () -> Int,
+    seekIcon: @Composable (modifier: Modifier) -> Unit
+) {
+    val isTapCountGreaterThanZero = tapCount > 0
+    Box(
+        modifier = modifier
+            .fillMaxHeight()
+            .indication(
+                indication = null,
+                interactionSource = remember { MutableInteractionSource() }
+            )
+            .pointerInput(tapCount) {
+                detectTapGestures(
+                    onTap = when (isTapCountGreaterThanZero) {
+                        true -> { _ -> onSeekSingleTap() }
+                        false -> { _ -> checkIfCanToggleIsControlsVisible() }
+                    },
+                    onDoubleTap = when (isTapCountGreaterThanZero) {
+                        true -> null
+                        false -> { _ -> onSeekDoubleTap() }
+                    }
+                )
```

Comment:
Since you only have `true`/`false` options, you can simplify it

```suggestion
onTap = if (isTapCountGreaterThanZero) {
                        { onSeekSingleTap() }
                    } else {
                        { checkIfCanToggleIsControlsVisible() }
                    },
                    onDoubleTap = if (isTapCountGreaterThanZero) {
                        null
                    } else {
                        { onSeekDoubleTap() }
                    },
```

Reply:
Is there a correct way in this case? I didn't use if/else because it seemed to me that with when it became more readable